### PR TITLE
SGX: QE identity info is now required to be available by the DCAP provider.

### LINKED ISSUE #2208: SGX: QE identity info is now required from the DCAP provider.
1. Previously if the quote provider did not return a valid QE identity info, the quote would be validated with a set of hard coded values.  Now it is required for the quote provider to return a valid QE identity info, this info is then used to verify the quote.
2. Fix doxygen documentation.

### PR DESCRIPTION
**Pull request by jazzybluesea**
1. For SGX quote verification, the QE identity info is now required.  Previously the QE identity info was not available in the DCAP provider and quote was verified against a set of hard-coded values.
2. Updated doxygen comments.

Fixes #2208

### CODE REVIEW

**Comment by jazzybluesea** [comment]
@johnkord  Still waiting for a new nuget package for the windows DCAP client to fix an issue in the dll.  This new package will need to get installed in all the windows systems.

File: tests/report/common/tests.cpp [hunk 1]
```diff
@@ -122,19 +123,18 @@ oe_result_t oe_get_collaterals(
...
- * @param collaterals_size The size of the **collaterals** buffer.
- * @param parsed_report Optional **oe_report_t** structure to populate with the
- * report properties in a standard format.
+ * @param[in] enclave The instance of the enclave that will be used.
+ * @param[in] report The buffer containing the report to verify.
+ * @param[in] report_size The size of the **report** buffer.
+ * @param[in] collaterals Optional The collateral data that is associated with
+ * the report.
+ * @param[in] collaterals_size The size of the **collaterals** buffer.
+ * @param[out] parsed_report Optional **oe_report_t** structure to populate with
```
Comment (gupta-ak) [inline tests/report/common/tests.cpp]: there's no comment for the the `input_validation_time` parameter.

**Comment by gupta-ak** [comment]
@jazzybluesea you can remove the "do not merge" label once the windows DCAP issue is fixed.

File: tests/report/common/tests.cpp (hunk 1 above)
Comment (jazzybluesea) [reply tests/report/common/tests.cpp]: Fixed.

**Comment by jazzybluesea** [comment]
> @jazzybluesea you can remove the "do not merge" label and do a bors r+ once the windows DCAP issue is fixed.

Yes, that's the plan.  CI/CD won't pass without the DCAP fix.

**Comment by johnkord** [comment]
@jazzybluesea Sounds good. Let me know when that new Azure DCAP Client nuget package is available and I'll make sure it gets deployed to all of the Windows CI environments!

**Comment by jazzybluesea** [comment]
@johnkord  The windows Azure DCAP 0.0.3 is now available.  Need to update the VMs for CI/CD to pass.  Thank you.

File: scripts/install-windows-prereqs.ps1 [hunk 2]
```diff
@@ -27,8 +27,8 @@ Param(
     [string]$IntelDCAPHash = 'F31E4451CA32E19CA3DCB0AFC49AFE9F4963C47BF62AAF24A8AE436BDA14FD8B',
     [string]$VCRuntime2012URL = 'https://download.microsoft.com/download/1/6/B/16B06F60-3B20-4FF2-B699-5E9B7962F9AE/VSU_4/vcredist_x64.exe',
     [string]$VCRuntime2012Hash = '681BE3E5BA9FD3DA02C09D7E565ADFA078640ED66A0D58583EFAD2C1E3CC4064',
-    [string]$AzureDCAPNupkgURL = 'https://www.nuget.org/api/v2/package/Azure.DCAP.Windows/0.0.2',
```
Comment (johnkord) [inline scripts/install-windows-prereqs.ps1]: Thanks for updating this!

**Comment by johnkord** [comment]
I've updated the CI systems to 0.0.3. Just ran a sandbox against master, and it looks good. Let's see how this branch does!

bors r+

**Comment by johnkord** [comment]
Okay, did a Resolve Conflicts :) Let's try that again

bors r+

**Comment by johnkord** [comment]
One more time! I think I fixed that one Linux system now, the one that caused the failure in the last bors run.

bors r+

**Comment by jazzybluesea** [comment]
@johnkord Thanks for taking care of this!

**Pull request merged**